File: Java/628_Maximum Product of Three Numbers.java
//Question - https://leetcode.com/problems/maximum-product-of-three-numbers/
class Solution {
    public int maximumProduct(int[] nums) {
        int n = nums.length;
        
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (nums[j] > nums[j + 1]) {
                    int temp = nums[j];
                    nums[j] = nums[j + 1];
                    nums[j + 1] = temp;
                }
            }
        }
         
        int op1 = (nums[0] * nums[1] * nums[n - 1]);
        int op2 = (nums[n - 1] * nums[n - 2] * nums[n - 3]);

        if(op1>op2)
        return op1;

        else
        return op2;

    }
}

File: C++/628_Maximum Product of Three Numbers.cpp
// Question - https://leetcode.com/problems/find-the-duplicate-number/

#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maximumProduct(vector<int> &nums)
    {
        sort(nums.begin(), nums.end()); // O(nlogn)
        return max((nums[0] * nums[1] * nums[nums.size() - 1]),
                   (nums[nums.size() - 1] * nums[nums.size() - 2] *
                    nums[nums.size() - 3]));
    }
};

int main()
{
    Solution obj;
    int n;
    cin >> n;
    vector<int> nums(n);
    for (int i = 0; i < n; i++)
    {
        cin >> nums[i];
    }
    int result = obj.maximumProduct(nums);
    cout << result << endl;
    return 0;
}
